ext/hook/uhook_legacy.c

```diff
@@ -306,36 +306,27 @@ static void dd_uhook(INTERNAL_FUNCTION_PARAMETERS, bool tracing, bool method) {
     zval *prehook = NULL, *posthook = NULL, *config_array = NULL;
     bool run_when_limited = false, allow_recursion = false;
 
-    ZEND_PARSE_PARAMETERS_START_EX(ZEND_PARSE_PARAMS_QUIET, 1 + method, 2 + method + !tracing)
+    ZEND_PARSE_PARAMETERS_START(1 + method, 2 + method + !tracing)
         // clang-format off
         if (method) {
             Z_PARAM_STR(class_name)
         }
         Z_PARAM_STR(method_name)
         Z_PARAM_OPTIONAL
-        if (!tracing) {
+
+        if (_i >= 0 && (uint32_t)_i == ZEND_NUM_ARGS()) {
```

Comment:
Why are we checking _i >= 0? Also maybe just:
```suggestion
        if (ZEND_NUM_ARGS() <= 1 + method)
```